test-evidence/src/main/scala/com/daml/test/evidence/generator/Main.scala

```diff
@@ -22,26 +24,37 @@ object Main {
     println(s"Wrote to $path")
   }
 
+  private val workspaceRoot: Option[String] = sys.env.get("BUILD_WORKSPACE_DIRECTORY")
+  private val targetDir = workspaceRoot.map(File(_)).getOrElse(File.currentWorkingDirectory)
+
   private def writeEvidenceToCsvFile[TE <: TestEntryCsv](
       fileName: String,
       entries: List[TE],
   ): Unit = {
     println(s"Writing inventory to $fileName...")
-    val file = File(fileName)
+    val file = targetDir / fileName
     val path = file.path.toAbsolutePath
     TestEntryCsvEncoder.write(file, entries)
     println(s"Wrote to $path")
   }
 
   def main(args: Array[String]): Unit = {
     if (args.length == 2) {
-      val securityTestEntries = TestEntryLookup.securityTestEntries
-      val csvEntries = securityTestEntries.map(SecurityTestEntryCsv.apply)
-      val csvFileName = args(0)
-      val jsonFileName = args(1)
+      val exitCode = Try {
+        val securityTestEntries = TestEntryLookup.securityTestEntries
+        val csvEntries = securityTestEntries.map(SecurityTestEntryCsv.apply)
+        val csvFileName = args(0)
+        val jsonFileName = args(1)
+
+        writeEvidenceToCsvFile(csvFileName, csvEntries)
+        writeEvidenceToJsonFile(jsonFileName, securityTestEntries)
+      } fold (e => {
```

Comment:
Yes, but I did not like this because of the literal numbers inside the different branches...
```scala
val exitCode = try {
  val securityTestEntries = TestEntryLookup.securityTestEntries
  val csvEntries = securityTestEntries.map(SecurityTestEntryCsv.apply)
  val csvFileName = args(0)
  val jsonFileName = args(1)

  writeEvidenceToCsvFile(csvFileName, csvEntries)
  writeEvidenceToJsonFile(jsonFileName, securityTestEntries)

  0
} catch {
  case NonFatal(e) =>
    e.printStackTrace()
    1
}

// explicitly exit, since some test suites have started non-daemon threads that prevent JVM shutdown
sys.exit(exitCode)
```
Alternatively, calling sys.exit twice:
```scala
try {
  val securityTestEntries = TestEntryLookup.securityTestEntries
  val csvEntries = securityTestEntries.map(SecurityTestEntryCsv.apply)
  val csvFileName = args(0)
  val jsonFileName = args(1)

  writeEvidenceToCsvFile(csvFileName, csvEntries)
  writeEvidenceToJsonFile(jsonFileName, securityTestEntries)

  // explicitly exit, since some test suites have started non-daemon threads that prevent JVM shutdown
  sys.exit(1)
} catch {
  case NonFatal(e) =>
    e.printStackTrace()
    // explicitly exit, since some test suites have started non-daemon threads that prevent JVM shutdown
    sys.exit(1)
}
```
WDYT?

Reply:
The literal numbers are pretty standard in scala (in java you would need to add `return`). 

Calling `sys.exit` twice is viable from my point of view.

In the end, it's your choice.